frappe/website/doctype/website_settings/google_indexing.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+import requests
+import googleapiclient.discovery
+import google.oauth2.credentials
+
+from frappe import _
+from googleapiclient.errors import HttpError
+from frappe.utils import get_request_site_address
+from frappe.model.document import Document
+from six.moves.urllib.parse import quote
+from frappe.integrations.doctype.google_settings.google_settings import get_auth_url
+
+SCOPES = "https://www.googleapis.com/auth/indexing"
+
+
+@frappe.whitelist()
+def authorize_access(reauthorize=None):
+	"""
+		If no Authorization code get it from Google and then request for Refresh Token.
+	"""
+
+	google_settings = frappe.get_doc("Google Settings")
+	website_settings = frappe.get_doc("Website Settings")
+
+	redirect_uri = get_request_site_address(True) + "?cmd=frappe.website.doctype.website_settings.google_indexing.google_callback"
+
+	if not website_settings.indexing_authorization_code or reauthorize:
+		return get_authentication_url(client_id=google_settings.client_id, redirect_uri=redirect_uri)
+	else:
+		try:
+			data = {
+				"code": website_settings.indexing_authorization_code,
```

Comment:
Rename `indexing_authorization_code` to `authorization_code` to avoid confusion

Reply:
why is it confusing? In future if we add more integrations, they might have their own authorization code. At that time having generic field name might be more confusing.